### ISSUE TITLE
Single API endpoint for uploading multiple scans/BOMs for a single project

### ISSUE THREAD

**Issue by eoftedal** [stevespringett/dependency-track#130]
For some projects I am using multiple scanners (OWASP Dependency check and retire.js and possibly also nsp). I would like to have the ability to upload multiple BOMs and scans in one go. The total would define the current state of the project, so if any dependencies are upgraded the old versions should be removed.

`PUT /api/v1/bomsandscans`
```
{ 
"project":"416ac246-e7ac-49ff-93b4-f7e94d997e6b",
"boms":[ ...base64encoded boms... ],
"scans":[ ...base64encoded scans... ]
}
```

**Comment by stevespringett** [comment]
Uploading a combination of boms and scans at the same time may not be doable, at least in the short term. Currently, you can upload multiple boms and scans on their respective endpoints, but at the time being, the artifacts are not processed in a single pass. That means, they will currently override each other. 

These endpoints both use POST and multi-part form data. So a JSON payload and Base64 encoding is not required. These endpoints will initially serve as the basis for the ability to manually upload artifacts through the UI and will later serve as a way to upload multiple artifacts (once the artifact merging logic is complete).

In short, this enhancement request is about 40% complete. The remaining work will be done at a later time.

**Comment by realHz** [comment]
We have kind of microservice architecture where a single project includes several 'services' (backend & ui) available in several repositories and built on different technologies (e.g. java mvn/gradle, javascript angular/react or similar).

For now we building each 'service', creating BOMs and upload them to dtrack which leads into suboptimal visualization in the dtrack ui where each 'subproject' (aka service) is listed as a project itself and its not possible to get a quick overview about the whole projects state.

i saw #571 and asked myself if it might be possible in future - or if there might be a workaround like 'merging BOMs' for example - to kind of consolidate different BOMs and visualize 'our terminology of a project' within one project in dtrack.

**Comment by stevespringett** [comment]
@realHz You may benefit from #84 once implemented. As a workaround, it's entirely possible to merge BOMs. Check out https://github.com/CycloneDX/cyclonedx-cli
